app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt

```diff
@@ -1,136 +1,71 @@
 package net.squanchy.favorites.view
 
 import android.content.Context
-import android.support.v7.widget.RecyclerView
+import android.support.v7.recyclerview.extensions.ListAdapter
+import android.support.v7.util.DiffUtil
 import android.view.LayoutInflater
 import android.view.ViewGroup
 import net.squanchy.R
 import net.squanchy.schedule.domain.view.Event
-import net.squanchy.schedule.domain.view.Schedule
-import net.squanchy.schedule.domain.view.SchedulePage
-import net.squanchy.schedule.view.EventItemView
-import net.squanchy.schedule.view.EventViewHolder
-import net.squanchy.search.view.HeaderViewHolder
-import org.joda.time.DateTimeZone
-import org.joda.time.LocalDate
 
-internal class FavoritesAdapter(context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+internal class FavoritesAdapter(
+    context: Context
+) : ListAdapter<FavoritesItem, FavoritesViewHolder<*>>(DiffCallback) {
```

Comment:
We have to give up the type parameter for now — such is the life when dealing with multiple view types and heterogeneous data, but we recover type safety in the actual implementation below